src/test/java/com/commercetools/sync/services/ProductServiceTest.java

```diff
@@ -0,0 +1,96 @@
+package com.commercetools.sync.services;
+
+import com.commercetools.sync.products.ProductTestUtils;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.products.Product;
+import io.sphere.sdk.products.ProductDraft;
+import io.sphere.sdk.products.commands.ProductUpdateCommand;
+import io.sphere.sdk.products.commands.updateactions.ChangeName;
+import io.sphere.sdk.products.commands.updateactions.Publish;
+import io.sphere.sdk.queries.PagedQueryResult;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Collections.singletonList;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class ProductServiceTest {
+
+    private ProductService service;
+    private SphereClient ctpClient;
+
+    @Before
+    public void setUp() {
+        ctpClient = spy(SphereClient.class);
+        service = ProductService.of(ctpClient);
+    }
+
+    @Test
+    public void create() {
+        when(ctpClient.execute(any())).thenReturn(completedFuture(mock(Product.class)));
+
+        Product product = service.create(mock(ProductDraft.class)).toCompletableFuture().join();
+
+        assertThat(product).isNotNull();
```

Comment:
I'm not sure this services should be covered by Unit tests - doesn't have much sense. 
But even if you decided mock it in unit test:

  1. 
```java
Product productMock = mock(Product.class);
when(ctpClient.execute(any())).thenReturn(completedFuture(productMock));
.......
assertThat(product).isSameAs(productMock);
```

  2. Since you are testing exactly `Create`:
```java
when(ctpClient.execute(any(ProductCreateCommand.class)))
......
verify(ctpClient).execute(any(ProductCreateCommand.class));
```